### PR TITLE
API guide updates

### PR DESCRIPTION
**Pull request by nickbianco**
This PR updates the API guide per discussion in #2254 and today's stand-up meeting.

It is still work-in-progress, but I'm creating the PR now in anyone wants to see the updates in the Doxygen build. I'll add reviewers once changes have been finalized.

### Brief summary of changes  API guide 
- Updated/added figures
- Moved over content from Confluence relevant to API structure
- Fixed markdown commands that weren't being parsed correctly.

### Looking for feedback on...
Overall layout, things to add/remove, general feedback, etc.

### CHANGELOG.md 
Should I update the change log?

The Doxygen for this PR can be viewed at http://myosin.sourceforge.net/?C=N;O=D; click the folder whose name is this PR's number.

---
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/2275)

### CODE REVIEW

**Comment by scrum2a** [comment]
@aymanhab asks should the guide be a single page? useful for printing?

**Comment by chrisdembia** [comment]
There is some content that you may be able to copy from https://simtk-confluence.stanford.edu:8443/display/OpenSim/OpenSim+Models

**Comment by chrisdembia** [comment]
@nickbianco I pushed a few updates.

**Comment by chrisdembia** [comment]
I'm up to "Main Categories of Components"

**Comment by chrisdembia** [comment]
I'm up to "Writing your own Component". I should be able to finish this pass later today.

**Comment by nickbianco** [comment]
Awesome. I like including the bicep curl example directly into the guide.

**Comment by chrisdembia** [comment]
I'm done making updates to the API Guide. 

The following sections of the API Guide are still mostly blank:

Joints

Joint
Constraints

Constraint
Forces

Force

PointToPointSpring
ContactForces
HuntCrossleyForce
ElasticFoundationForce
Actuators

Actuator

CoordinateActuator
Muscle
Controllers

Controller


@nickbianco and I discussed these sections briefly; they might not even be necessary.

**Comment by chrisdembia** [comment]
> Should I update the change log?

No I don't think so.

**Comment by nickbianco** [comment]
@aseth1, Chris and I have been making additions/changes to the API here too. I've added you as a reviewer  since you we're planning to look over API guide changes per the scrum meeting today.

**Comment by nickbianco** [comment]
@aseth1 @chrisdembia this is ready for review.

**Comment by chrisdembia** [comment]
The italics seem to be hit or miss:

image


The caption for Figure 3 is not on its own line:

image


Other than these two items, this PR looks good.

**Comment by nickbianco** [comment]
@aseth1 ready for review.

File: doc/APIGuide.md [hunk 1]
```diff
@@ -19,21 +19,22 @@ The ethos of the OpenSim API is that the code is modular, reusable, and easily e
 
 ## Organization of OpenSim {#organization}
 
-OpenSim is built on the computational and simulation core provided by SimTK. This core includes low-level, efficient math and matrix algebra libraries, such as LAPACK, as well as the infrastructure for defining a dynamic System and its State. One can think of the System as the set of differential equations, and the state as its variables. 
+OpenSim is built on the computational and simulation libraries provided by SimTK. This core includes low-level, efficient math and matrix algebra libraries, such as LAPACK, as well as the infrastructure for defining a dynamic System and its State. One can think of the System as the set of differential equations, and the state as its variables.
```
Comment (aseth1) [inline doc/APIGuide.md]: Consider:

OpenSim is built on SimTK computational and simulation libraries that provide low-level, ...

Comment (aseth1) [reply doc/APIGuide.md]: No comma after low-level either.

**Pull request merged**